feat: Add chart for storage per broker

hemahg opened the pull request:
Signed-off-by: hemahg <hhg@redhat.com>

**What this PR does / why we need it**:
- [ ] Add filter to filter by Broker
- [ ]  Add toggle filter for broker.
- [ ]  Add a chart to show storage per instance
- [ ]  Add a chart to show storage per broker

**Which issue(s) this PR fixes** 
fixes https://issues.redhat.com/browse/MGDX-258

> Replace this comment with references to related issues.
> e.g. `fixes #<issue number>`

**Verification steps** 

> Attach a screenshot of the output.
> guidance on how to go the particular story.
> e.g. `point to the story with a text breadcrumb, something like "components > <component name> > <story name>”`

**Is there any work left / what's next** :

> Replace this comment with pending items that need to be covered in this PR or some pending items you are planning to do in the future PR.

**Special notes for your reviewer**:

hemahg commented:
I am working on integrating UI with backend, I will merge this PR when that is done

riccardo-forina commented on packages/consoledot-api/src/fetchers/fetchKafkaMetrics.ts:
```diff
@@ -54,9 +73,33 @@ export async function fetchKafkaMetrics({
     }
 
+    function addAggregateBrokerBytes() {
+      console.log(broker_id);
```
Rogue log?

riccardo-forina commented on packages/consoledot-api/src/fetchers/fetchKafkaMetrics.ts:
```diff
@@ -54,9 +73,33 @@ export async function fetchKafkaMetrics({
+          (broker[timestamp] = value + (broker[timestamp] || 0))
+      );
+      usedDiskSpaceMetrics[broker_id] = broker;
+      console.log(broker);
```
Another one

riccardo-forina commented on packages/consoledot-api/src/fetchers/fetchKafkaMetrics.ts:
```diff
@@ -54,9 +73,33 @@ export async function fetchKafkaMetrics({
+    function addAggregateBrokerBytes() {
+      console.log(broker_id);
+      const broker =
+        usedDiskSpaceMetrics[broker_id] || usedDiskSpaceMetrics["2"];
```
Can we have a comment to describe this magic "2" value? It's not clear to me why we have it by looking at this.

riccardo-forina reviewed (approved):
I think this is ok? I haven't tested it tho.

Left a few comments here and there.

hemahg commented on packages/consoledot-api/src/fetchers/fetchTopicsMetrics.ts:
```diff
@@ -53,6 +55,7 @@ export async function fetchTopicsMetrics({
   const incomingMessageRate: TimeSeriesMetrics = {};
 
   filteredMetrics.forEach((m) => {
+    console.log(m);
```
apologize @riccardo-forina, I am still working on this. I will remove the logs

hemahg commented on packages/ui/src/components/Metrics/components/ChartPartitionSizePerBroker.tsx:
```diff
@@ -0,0 +1,190 @@
+    const showDate = shouldShowDate(duration);
+
+    return (
+      <div ref={containerRef} style={{ marginTop: "-30px", height: "700px" }}>
```
Hi @jgiardino, I have fixed the legend name visible issue by adjusting the height and 
 [This](https://63c984f3273772562ae72406-ymjdmvpxrf.chromatic.com/?path=/story/components-metrics--all-ready ) is how the charts look, could you please have a look, if the topic length is more again some of the names of the partitions may not be visible, please let me know your opinion on this

jgiardino commented:
@hemahg - Thanks for making updates to this!

One easy update that's needed is to change the format of legend labels to include extra spaces around the slash "/". This is what we have now:
`1, test-topic-1/0`
but it should be this instead:
`1, test-topic-1 / 0`

As for the layout and responsive behavior of the chart, this is what I noticed:
1. **Ability to test responsiveness of legend for varying lengths in legend labels -** It's difficult to tell how responsive the legend is with the example provided. It would be helpful to be able to modify the topic names so that we can assess how responsive the legend is depending on variable topic name lengths. I did a quick test of this by modifying the label via the web inspector, and noticed that the legend was not responsive to a longer length as shown in the first example below, but I'm not sure if that's a valid test of this. 
2. **Overall layout of chart within card body -**  There is extra space above the Partition size charts that shouldn't be there. I don't see this extra space in any other chart. 
3. **Legend area has a fixed height -** While the legend labels wrap as expected, the section where the legend displays doesn't change in height to accomodate the height of the contents. You can see this behavior in the examples below. The last example illustrates how some labels aren't visible at all because the content height is greater than the container height.

It would be good for you to reach back out on PF slack to get a review of the issues you're struggling with and further advice on how to address the lack of responsiveness that we're setting. The legend should allow for any labels with variable length and any height in overall contents. The legend should do this by wrapping the labels as needed (accounting for variable label lengths) and allowing the height to grow and shrink as needed. 

Before reaching out, it might be helpful to enable the example so that we can properly test the label length as noted in point 1. above. 

<img width="655" alt="image" src="https://user-images.githubusercontent.com/21063328/219870597-ad8bf15d-5179-4ff1-93d6-27dff558fde9.png">

<img width="951" alt="image" src="https://user-images.githubusercontent.com/21063328/219870976-8b2534cf-6bdc-4885-b230-7d5010611de1.png">

<img width="453" alt="image" src="https://user-images.githubusercontent.com/21063328/219871010-dec77a2c-16bf-40fd-a1f7-d5f06cedeed0.png">

hemahg commented:
Hi @jgiardino, I have updated the legend labels from `1, test-topic-1/0` to  `1, test-topic-1 / 0`
1. The only way I can display All the legend labels is by increasing the height in the div tag `<div ref={containerRef} style={{margin-top: "-30px", height: "700px", width: "650px" }}>` if the length of the topic name is long I need to increase the height prop in the div tag. 
2. There is extra space above the Partition size charts that shouldn't be there.  The extra space is because of the increase in the height in the div tag, I can decrease the spacing above the Partition size charts by changing the marginTop props in the div tag, but it’s making the partition toggle not functional, I can't toggle the partition select toggle
![image](https://user-images.githubusercontent.com/53568062/220260356-aab246e2-dd05-4ca5-8f9c-38778c73f279.png) These are the two problems with displaying legend labels

I have updated the stories https://63c984f3273772562ae72406-uomorypctb.chromatic.com/?path=/story/components-metrics--all-ready to include different topic names, with variable lengths. please let me know if this example is good enough to showcase this problem in PF slack
